test/optim/test_optimize.py

```diff
@@ -546,9 +655,32 @@ def test_optimize_acqf_batch_limit(self) -> None:
                     gen_candidates=gen_candidates,
                     batch_initial_conditions=ics,
                 )
-                expected_shape = (num_restarts,) if ics is None else (ics.shape[0],)
+                expected_shape = (
+                    (num_restarts,)
+                    if (ics is None) or (ics.shape[-2] < raw_samples)
+                    else (ics.shape[0],)
+                )
                 self.assertEqual(acq_value_list.shape, expected_shape)
 
+        with self.subTest(gen_candidates=gen_candidates):
+            with self.assertWarnsRegex(
+                DeprecationWarning, "Defaulting to old behavior"
+            ):
+                ics = torch.ones((2, 1, dim))
+                _, acq_value_list = optimize_acqf(
+                    acq_function=SinOneOverXAcqusitionFunction(),
+                    bounds=torch.stack([-1 * torch.ones(dim), torch.ones(dim)]),
+                    q=q,
+                    num_restarts=num_restarts,
+                    raw_samples=raw_samples,
+                    options=options,
+                    return_best_only=False,
+                    gen_candidates=gen_candidates,
+                    batch_initial_conditions=ics,
+                )
+
+                self.assertEqual(acq_value_list.shape, (ics.shape[0],))
+
     def test_optimize_acqf_runs_given_batch_initial_conditions(self):
         num_restarts, raw_samples, dim = 1, 2, 3
```

Comment:
GH won't let me comment there, but updating the second tensor definitions below from `torch.tensor([1])` to `torch.tensor([1.])` fixes that issue. I believe it expects the coefficients in the constraints to be floats. The test still fails with a shape error when `shape=(1, dim)`.

Reply:
thanks, I'll look at the other test case issue tomorrow. I was trying to understand what I had done that was changing dtypes but if this is okay then will just update the test accordingly

Reply:
I don't think you changed any dtypes. Previously, it would've just used the given ICs and not called `ic_generator` with those constraints (with wrong dtypes), since it didn't need to generate more. The scipy optimizer uses a different format for the constraints, so they're modified before passing in. It just wasn't executing the code path (polytope sampler) that triggered the error.

Reply:
```
            if (
                self.raw_samples is not None
                and (self.raw_samples - batch_initial_conditions_shape[-2]) > 0
                and len(batch_initial_conditions_shape) == 3
                and self.num_restarts is not None
                and batch_initial_conditions_shape[0] not in [1, self.num_restarts]
            ):
```

this logic should have been catching the case but wasn't, will fix